packages/jsActions/nanoflow-actions-native/src/geolocation/GetCurrentLocation.ts

```diff
@@ -38,20 +40,35 @@ export async function GetCurrentLocation(
 
     return new Promise((resolve, reject) => {
         const options = getOptions();
-        navigator.geolocation.getCurrentPosition(onSuccess, onError, options);
+        const watchId: number = navigator.geolocation.watchPosition(onSuccess, onError, options);
+        const timeStart = Date.now();
+        let lastAccruedPosition: GeolocationResponse;
 
-        function onSuccess(position: GeolocationResponse): void {
+        function createGeolocationObject(position: GeolocationResponse): void {
             mx.data.create({
                 entity: "NanoflowCommons.Geolocation",
-                callback: mxObject => {
-                    const geolocation = mapPositionToMxObject(mxObject, position);
-                    resolve(geolocation);
-                },
+                callback: mxObject => resolve(mapPositionToMxObject(mxObject, position)),
                 error: () =>
                     reject(new Error("Could not create 'NanoflowCommons.Geolocation' object to store location"))
             });
         }
 
+        function onSuccess(position: GeolocationResponse): void {
+            if (watchId && (!minimumAccuracy || minimumAccuracy >= position.coords.accuracy)) {
+                navigator.geolocation.clearWatch(watchId);
+                createGeolocationObject(position);
+            } else {
+                if (!lastAccruedPosition || position.coords.accuracy < lastAccruedPosition.coords.accuracy) {
+                    lastAccruedPosition = position;
+                }
+                const timeDiff = Date.now() - timeStart;
+                if (!timeout || timeout.lte(timeDiff)) {
+                    navigator.geolocation.clearWatch(watchId);
+                    createGeolocationObject(lastAccruedPosition);
```

Comment:
Should `createGelocationObject` here not be called with `position` instead of `lastAccruedPosition` as that's the more recent one? Or do you want to higher quality one?

Reply:
the lastAccruedPosition is the most updated object, so in this case, nope.

Reply:
Actually lastAccruedPosition is the latest, so in this case only timeout is checked and then we retrieve the last updated one.